tests/test_create.py

```diff
@@ -2177,94 +2177,105 @@ def side_effect(self, url, **kwargs):
+        assert package_is_installed(prefix, "conda")
+        assert package_is_installed(prefix, "pycosat")
+        assert package_is_installed(prefix, "conda-build")
 
 
-def test_dont_remove_conda_2():
+def test_dont_remove_conda_2(
+    conda_cli: CondaCLIFixture, tmp_env: TmpEnvFixture, monkeypatch: MonkeyPatch
+):
     # regression test for #6904
-    pkgs_dirs = context.pkgs_dirs
-    prefix = make_temp_prefix()
-    with make_temp_env(prefix=prefix):
-        with env_vars(
-            {"CONDA_ROOT_PREFIX": prefix, "CONDA_PKGS_DIRS": ",".join(pkgs_dirs)},
-            stack_callback=conda_tests_ctxt_mgmt_def_pol,
-        ):
-            _, _, _ = run_command(Commands.INSTALL, prefix, "conda")
-            assert package_is_installed(prefix, "conda")
-            assert package_is_installed(prefix, "pycosat")
+    with tmp_env() as prefix:
+        monkeypatch.setenv("CONDA_ROOT_PREFIX", prefix)
+        reset_context()
+        assert context.root_prefix == str(prefix)
 
-            with pytest.raises(CondaMultiError) as exc:
-                run_command(Commands.REMOVE, prefix, "pycosat")
+        _, _, _ = conda_cli("install", f"--prefix={prefix}", "conda", "--yes")
+        assert package_is_installed(prefix, "conda")
+        assert package_is_installed(prefix, "pycosat")
+
+        with pytest.raises(CondaMultiError) as exc:
+            conda_cli("remove", f"--prefix={prefix}", "pycosat", "--yes")
 
-            assert any(isinstance(e, RemoveError) for e in exc.value.errors)
-            assert package_is_installed(prefix, "conda")
-            assert package_is_installed(prefix, "pycosat")
+        assert any(isinstance(e, RemoveError) for e in exc.value.errors)
+        assert package_is_installed(prefix, "conda")
+        assert package_is_installed(prefix, "pycosat")
 
-            with pytest.raises(CondaMultiError) as exc:
-                run_command(Commands.REMOVE, prefix, "conda")
+        with pytest.raises(CondaMultiError) as exc:
+            conda_cli("remove", f"--prefix={prefix}", "conda", "--yes")
 
-            assert any(isinstance(e, RemoveError) for e in exc.value.errors)
-            assert package_is_installed(prefix, "conda")
-            assert package_is_installed(prefix, "pycosat")
+        assert any(isinstance(e, RemoveError) for e in exc.value.errors)
+        assert package_is_installed(prefix, "conda")
+        assert package_is_installed(prefix, "pycosat")
 
 
-def test_force_remove():
-    with make_temp_env() as prefix:
-        stdout, stderr, _ = run_command(Commands.INSTALL, prefix, "libarchive")
+def test_force_remove(
+    tmp_env: TmpEnvFixture, conda_cli: CondaCLIFixture, monkeypatch: MonkeyPatch
+):
+    with tmp_env() as prefix:
+        monkeypatch.setenv("CONDA_ROOT_PREFIX", prefix)
+        reset_context()
+        stdout, stderr, _ = conda_cli(
+            "install", f"--prefix={prefix}", "libarchive", "--yes"
+        )
```

Comment:
we don't need `CONDA_ROOT_PREFIX` for this test and we can combine the env creation and package installation:

```suggestion
    with tmp_env("libarchive") as prefix:
```